src/value.rs

```diff
@@ -54,21 +84,50 @@ impl NixValue {
     pub fn format_markdown(&self) -> String {
         use NixValue::*;
         match self {
-            Bool(x) => format!("```nix\n{}\n```", x),
-            Float(x) => format!("```nix\n{}\n```", x),
-            Integer(x) => format!("```nix\n{}\n```", x),
+            Str(_) | Bool(_) | Float(_) | Integer(_) | Null => format!("```nix\n{}\n```", self),
+            Map(_) => format!("```text\n{}\n```", self),
         }
     }
 }
 
-impl Debug for NixValue {
+/// Maximum number of keys to show when displaying attribute sets in tooltips.
+const MAX_DISPLAY_KEYS: usize = 5;
+
+impl Display for NixValue {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
         match self {
+            NixValue::Str(x) => write!(f, "{:?}", x),
             NixValue::Bool(x) => write!(f, "{}", x),
-            // TODO: format nicely like `nix rep`
+            // TODO: format nicely like `nix repl`
             NixValue::Float(x) => write!(f, "{}", x),
             NixValue::Integer(x) => write!(f, "{}", x),
+            NixValue::Null => write!(f, "null"),
+            NixValue::Map(map) => {
+                let mut keys = map
+                    .keys()
+                    .take(MAX_DISPLAY_KEYS)
```

Comment:
Maybe we should just sort all keys? They're already stored in the hashmap, so this shouldn't be too expensive?

Reply:
IIRC this is only relevant for rendering where we only show 5 keys, correct? Because of that I considered it rather low-impact, but if you think it makes sense here, then go for it :)

Reply:
Yep, that's correct. I think it probably makes sense to leave as-is, since we don't want the UI to slow down on massive attribute sets